sunpy/image/transform.py

```diff
@@ -126,8 +130,27 @@ def affine_transform(image, rmatrix, order=3, scale=1.0, image_center=None,
                           SunpyUserWarning)
             adjusted_image = np.nan_to_num(adjusted_image)
 
+        # Scale image to range [0, 1] if it is valid (not made up entirely of NaNs)
+        is_nan_image = np.all(np.isnan(adjusted_image))
+        if is_nan_image:
+            adjusted_missing = missing
+        else:
+            im_min = np.nanmin(adjusted_image)
+            adjusted_image -= im_min
+            im_max = np.nanmax(adjusted_image)
+            if im_max > 0:
+                adjusted_image /= im_max
+                adjusted_missing = (missing - im_min) / im_max
+            else:
+                adjusted_missing = missing - im_min
+
         rotated_image = skimage.transform.warp(adjusted_image, tform, order=order,
-                                               mode='constant', cval=missing,
-                                               preserve_range=True)
+                                               mode='constant', cval=adjusted_missing, preserve_range=True)
```

Comment:
```suggestion
                                               mode='constant', cval=adjusted_missing)
```
Given that we're doing the keyword `preserve_range` applies only to passing in an integer array, and our code never does that, it's probably best to remove this non-functional keyword argument.